project-koku/koku
# [COST-3714] Check if ROS report is in current payload before trying to process

## Jira Ticket

[COST-3714](https://issues.redhat.com/browse/COST-3714)

## Description

In prod, we have some customers that generate large enough files that the payloads are split for one manifest. This change will check if a ROS file is in the current payload before attempting to process it, allowing ROS files from different payloads to process properly.

## Testing
Reach out if you need help with these steps or an example report from the operator to send.
1. Reserve an ephemeral namespace
2. Deploy main to an ephemeral namespace
3. Alter the manifest from an operator report to have multiple `resource_optimization_files` even though there is only one present.
4. Create a source in ephemeral that matches the cluster in your operator report.
5. Send the payload to ingress for the ephemeral namespace and verify that the ROS report processing fails. You will see a message similar to: 
```
[2023-04-19 12:10:11,965] WARNING None {'message': "ROS reports not processed for manifest_id 1. Reason: [Errno 2] No such file or directory: '/var/tmp/masu/tmpeee2adc6/20230330T141336-cost-mgmt/795eb333-370a-4653-aa40-0f7fd911b61b_openshift_usage_report.5.csv'", 'tracing_id': '795eb333-370a-4653-aa40-0f7fd911b61b', 'account': '10001', 'org_id': '1234567', 'cluster_id': '023d9b0e-7ca6-481d-b04f-ea606becd54e', 'provider_uuid': 'd8367573-f065-42da-ad2d-62e348e2e356', 'provider_type': 'OCP'}

```
6. Deploy this branch to the ephemeral namespace you reserved.
7. Try to send that same payload to ingress for the ephemeral namespace, this time the ROS report processing should succeed and you should see a ROS report get processed.
8. If you try and send a payload that does not contain a ROS report but there are ROS reports in the manifest, you should see something similar to: 
```
[2023-04-19 12:20:26,569] INFO None {'message': 'No ROS reports to handle in the current payload.', 'tracing_id': 'ingress-service-7cbdc585ff-nvl7g/feGyFqWZD6-000685', 'account': '10001', 'org_id': '1234567', 'cluster_id': '023d9b0e-7ca6-481d-b04f-ea606becd54e', 'provider_uuid': 'd8367573-f065-42da-ad2d-62e348e2e356', 'provider_type': 'OCP', 'manifest_id': 1}

```
## Notes

...
